service_analysis/app/repositories/dao/AnalysisDao.scala

```diff
@@ -0,0 +1,206 @@
+import play.api.db.slick.DatabaseConfigProvider
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json.Json
+import slick.dbio.DBIOAction
+import slick.dbio.Effect.All
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+@Singleton
+class AnalysisDao @Inject() (
+    val dbConfigProvider: DatabaseConfigProvider
+) extends Tables {
+
+  val logger = Logger(classOf[AnalysisDao])
+
+  import driver.api._
+
+  private val noaction: DBIO[Unit] = DBIO.successful(())
+
+  private def insertAnalysisAction(event: EventRow): DBIO[EventId] = {
+    analysisTable returning analysisTable.map(_.id) += event
+  }
+
+  private def insertResultAction(result: ResultRow): DBIO[Long] = {
+    resultTable returning resultTable.map(_.id) += result
+  }
+
+  private def insertAnalysisWithResultAction(a: Analysis): DBIO[EventId] = {
+    for {
+      id <- insertAnalysisAction(asEventTuple(a))
+      _ <- a.result.map(r => insertResultAction(asResultTuple(id, r))).getOrElse(noaction)
+    } yield id
+  }
+
+  private def insertChildEventsAction(
+    pid: EventId,
+    events: Seq[Analysis]
+  ): DBIO[Seq[EventId]] = {
+    val batch = events.map { e =>
+      insertAnalysisWithResultAction(e.copy(partOf = Some(pid)))
+    }
+    DBIO.sequence(batch)
+  }
+
+  private def findByIdAction(id: EventId): DBIO[Option[AnalysisEvent]] = {
+    analysisTable.filter(_.id === id).result.headOption.map { res =>
+      res.flatMap(fromEventRow)
+    }
+  }
+
+  private def resultForEventIdAction(id: EventId): DBIO[Option[AnalysisResult]] = {
+    resultTable.filter(_.eventId === id).result.headOption.map { res =>
+      res.flatMap(fromResultRow)
+    }
+  }
+
+  private def listChildrenAction(parentId: EventId): DBIO[Seq[Analysis]] = {
+    val query = analysisTable.filter(_.partOf === parentId) joinLeft
+      resultTable on (_.id === _.eventId)
+
+    query.result.map { res =>
+      res.map { row =>
+        fromEventRow(row._1)
+          .flatMap(_.withResultAsOpt(fromResultRow(row._2))).get
+      }
+    }
+  }
+
+  private def listForObjectAction(oid: ObjectUUID): DBIO[Seq[AnalysisEvent]] = {
+    val query = analysisTable.filter(_.objectUuid === oid) joinLeft
+      resultTable on (_.id === _.eventId)
+
+    query.result.map { res =>
+      res.flatMap { row =>
+        fromEventRow(row._1).map { ue =>
+          ue.withResultAsOpt(fromResultRow(row._2)).getOrElse(ue)
+        }
+      }
+    }
+  }
+
+  /**
+   * Write a single {{{Analysis}}} event to the DB.
+   *
+   * @param a The Analysis to persist.
+   * @return eventually returns a MusitResult containing the EventId.
+   */
+  def insert(a: Analysis): Future[MusitResult[EventId]] = {
+    val action = insertAnalysisWithResultAction(a)
+
+    db.run(action.transactionally).map(MusitSuccess.apply).recover {
+      case NonFatal(ex) =>
+        val msg = s"An unexpected error occurred inserting an analysis event"
+        logger.error(msg, ex)
```

Comment:
Have we defined any guidelines for logging? Imo this kind of faults should be logged as warning.